src/group.rs

```diff
@@ -32,52 +42,100 @@ impl GroupCreateOpts {
     /// # Arguments
     /// - `id` - Unique id of a group within a segment. If none, the server will assign an id.
     /// - `name` - Human readable name of the group. Does not need to be unique. Will **not** be encrypted.
+    /// - `add_as_admin`
+    ///     - true (default) - The creating user will be added as an admin of the group.
+    ///     - false - The creating user will not be an admin of the group.
     /// - `add_as_member`
-    ///     - `true` - The creating user should be added as a member (in addition to being a group admin)
-    ///     - `false` - The creating user will not be a member of the group, but will still be an admin.
-    /// - `members` - List of users to be added as members of the group.
-    ///               Note: even if `add_as_member` is false, the calling user will be added as a member if they are in this list.
+    ///     - true (default) - The creating user will be added as a member of the group.
+    ///     - false - The creating user will not be a member of the group.
+    /// - `owner` - Specifies the owner of the group
+    ///     - None (default) - The creating user will be the owner of the group. Cannot be used if `add_as_admin` is set to false as the owner must be an admin.
+    ///     - Some(UserId) - The provided user will be the owner of the group. This ID will automatically be added to the admins list.
+    /// - `admins` - List of users to be added as admins of the group. This list takes priority over `add_as_admin`,
+    ///             so the calling user will be added as a member if their id is in this list even if `add_as_admin` is false.
+    /// - `members` - List of users to be added as members of the group. This list takes priority over `add_as_member`,
+    ///             so the calling user will be added as a member if their id is in this list even if `add_as_member` is false.
     /// - `needs_rotation`
     ///     - true - group's private key will be marked for rotation
     ///     - false (default) - group's private key will not be marked for rotation
     pub fn new(
         id: Option<GroupId>,
         name: Option<GroupName>,
+        add_as_admin: bool,
         add_as_member: bool,
+        owner: Option<UserId>,
+        admins: Vec<UserId>,
         members: Vec<UserId>,
         needs_rotation: bool,
     ) -> GroupCreateOpts {
         GroupCreateOpts {
             id,
             name,
+            add_as_admin,
             add_as_member,
+            owner,
+            admins,
             members,
             needs_rotation,
         }
     }
 
-    fn standardize(self, calling_id: &UserId) -> GroupCreateOpts {
+    pub(crate) fn standardize(self, calling_id: &UserId) -> Result<GroupCreateOptsStd> {
```

Comment:
Why is this pub(crate)?

Reply:
I don't remember. I added it at some point when it said the function was private and I needed it, but I assume moving things around fixed that? So I've removed it now